src/proof/proof_node_updater.cpp

```diff
@@ -203,12 +210,21 @@ void ProofNodeUpdater::processInternal(std::shared_ptr<ProofNode> pf,
       // proof with the same result. Same as above, updating the contents here
       // is typically not necessary since references to this proof will be
       // replaced.
-      if (checkMergeProof(cur, resCache, cfaMap))
+      // maybe found a proof in the meantime, i.e. a subproof of the current
+      // proof with the same result. Same as above, updating the contents here
+      // is typically not necessary since references to this proof will be
+      // replaced.
+      if (!checkMergeProof(cur, resCache, cfaMap))
       {
-        visited[cur] = true;
```

Comment:
What happened to this line?

Reply:
It was redundant, it was already set earlier in the method.